apps/project/schema.py

```diff
@@ -246,6 +248,42 @@ def resolve_regions(root, info, **kwargs):
         return info.context.dl.project.geo_region.load(root.pk)
 
 
+class AnalysisFrameworkForPublicProjectType(DjangoObjectType):
+    class Meta:
+        model = AnalysisFramework
+        skip_registry = True
+        fields = (
+            'id', 'title'
+        )
+
+
+class PublicProjectType(DjangoObjectType):
+    class Meta:
+        model = Project
+        skip_registry = True
+        fields = (
+            'id', 'description', 'title',
+            'created_at'
+        )
+    stats = graphene.Field(ProjectStatType)
+    regions = DjangoListField(RegionDetailType)
+    organizations = graphene.List(graphene.NonNull(ProjectOrganizationType))
+    analysis_framework = graphene.Field(AnalysisFrameworkForPublicProjectType)
+
+    @staticmethod
+    def resolve_organizations(root, info):
+        return info.context.dl.project.organizations.load(root.pk)
+
+    def resolve_regions(root, info, **kwargs):
+        # Need to have a base permission
+        if PP.check_permission(info, PP.Permission.BASE_ACCESS):
+            return info.context.dl.project.geo_region.load(root.pk)
+        return info.context.dl.project.geo_region.load(root.pk)
```

Comment:
Can you change this for ProjectType as well so that for base all are return and for other only public are returned.

You can pass (root.pk, True/False), and filter-out private for False. OR use separate dataloaders.

For publicprojecttype filter out private regions only.